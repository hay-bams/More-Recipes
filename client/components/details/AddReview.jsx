import React from 'react';

const AddReview = () => (
  <div className="container">
    <div className="row review-form">
      <div className="col-sm-6">
        <h4>Add Review</h4>
        <form>
          <div className="form-group">
            <textarea className="form-control" />
          </div>

          <div className="form-group">
            <input type="submit" className="btn btn-success" />
          </div>
        </form>
      </div>
    </div>
  </div>
);

export default AddReview;
