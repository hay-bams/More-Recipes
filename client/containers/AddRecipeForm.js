import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux'; 
import { Link } from 'react-router-dom';
import { addRecipe } from '../actions/actions.js';
import { clearError } from '../actions/actions.js';

/**
 * @class AddRecipeForm
 */
class AddRecipeForm extends React.Component {
  constructor(props) {
    super(props);
    this.addRecipe =  this.addRecipe.bind(this);
    this.recipeNum = this.props.recipes.length;
  }

  componentWillMount() {
     this.props.clearError()
  }

  addRecipe(event) {
    event.preventDefault();
    const recipe = {
      title: this.title.value,
      ingredients: this.ingredients.value,
      instructions: this.instructions.value,
      image: 'imageURl'
    }
    this.props.addRecipe(recipe);
  } 

  render() {
    return (
      <div className = "main-userboard-body">
        <div className = "container-fluid">
          <ol className="breadcrumb">
            <li className="breadcrumb-item">
              <a href="#">Dashboard</a>
            </li>
            <li className="breadcrumb-item active">Add Recipe</li>
          </ol>
        </div>
  
        <div className = "container">
          <div className = "row">
            <div className = "col-sm-12 col-md-8 col-lg-6">
           {this.recipeNum = this.props.recipes.length - this.recipeNum === 1 ? 
            this.props.clearError() && <span></span> : ""} 
              <span>{this.props.message.errors ? this.props.message.errors.response.data.message: ""}</span>
                <form onSubmit = {this.addRecipe}>
                <div className = "form-group">
                  <label htmlFor = "food">Recipe Name</label>
                  <input ref = {(input) => this.title = input} type="text" name="title" 
                  className = "form-control" placeholder="Enter Recipe name" />
                </div>
  
                <div className = "form-group">
                  <input type="file" />
                </div>
  
                  <div className = "form-group">
                  <label htmlFor = "ingredient">Ingredients</label>
                  <input  ref = {(input) => this.ingredients = input}ype="text" name="ingredient" 
                  className = "form-control" placeholder="Enter Ingredients" />
                </div>
  
                <div className = "form-group">
                  <label htmlFor = "instruction">Instructions</label>
                  <textarea ref = {(input) => this.instructions = input} className = "form-control" 
                  placeholder="Enter Instructions" name = "instruction"></textarea>
                </div>
  
                <div className = "form-group">
                  <input type="submit" className = "btn btn-success" value = "Add Recipe" />
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
  
    )
  }
}

const mapStateToProps = state => {
  return {
    recipes: state.recipes,
    message: state.errors
  }
}

const mapDispatchToProps = dispatch => {
  return bindActionCreators({addRecipe, clearError}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(AddRecipeForm);