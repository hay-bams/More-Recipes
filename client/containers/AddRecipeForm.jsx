import React from 'react';
import axios from 'axios';
import Dropzone from 'react-dropzone';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import Loader from 'react-loader';
import { addRecipe } from '../actions/recipe';
import Authenticate from '../auth/auth';

const dropZoneStyles = {
  border: 'none',
  cursor: 'pointer'
};

/**
 * @class AddRecipeForm
 *
 */
export class AddRecipeForm extends React.Component {
  /**
   * @param {obj} props
   * @returns {void} constructor
   */
  constructor(props) {
    super(props);
    this.addRecipe = this.addRecipe.bind(this);
    this.uploadImageToCloudinary = this.uploadImageToCloudinary.bind(this);
    this.handleDrop = this.handleDrop.bind(this);
    this.onChange = this.onChange.bind(this);
    this.state = {
      imageURI: null,
      image: null,
      loaded: true,
      uploadError: null,
      ingredientError: '',
      titleError: '',
      instructionError: '',
      addingRecipe: false
    };
  }

  /**
   * @param {obj} event
   * @returns {void} onChange
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   *
   * @param {obj} file
   * @return {void} handleDrop
   */
  handleDrop(file) {
    this.setState({ image: file[0] });
  }

  /**
   *
   * @param {obj} event
   * @return {obj} uploadImageToCloudinary
   */
  async uploadImageToCloudinary() {
    try {
      const imageFile = this.state.image;
      const CLOUDINARY_URL = 'https://api.cloudinary.com/v1_1/dsj9ygnq2/upload';
      const CLOUDINARY_UPLOAD_PRESET = 'xmklgrkm';
      const formData = new FormData();
      formData.append('file', imageFile);
      formData.append('upload_preset', CLOUDINARY_UPLOAD_PRESET);
      const response = await axios({
        url: CLOUDINARY_URL,
        method: 'POST',
        headers: { 'X-Requested-With': 'XMLHttpRequest' },
        data: formData
      });
      return Promise.resolve(response.data.secure_url);
    } catch (err) {
      return Promise.reject(err);
    }
  }

  /**
   * @param {obj} event
   * @returns {void} addRecipe
   */
  async addRecipe(event) {
    try {
      event.preventDefault();
      let recipe = {
        title: this.state.title,
        ingredients: this.state.ingredients,
        instructions: this.state.instructions,
      };

      const errors = Authenticate.validateAddRecipe(recipe);

      if (errors.title !== '' || errors.ingredients !== '' ||
     errors.instructions) {
        return this.setState({
          ingredientError: errors.ingredients,
          titleError: errors.title,
          instructionError: errors.instructions
        });
      }
      this.setState({ loaded: false, addingRecipe: true });
      const imageURI = await this.uploadImageToCloudinary();

      recipe = {
        title: this.state.title,
        ingredients: this.state.ingredients,
        instructions: this.state.instructions,
        image: imageURI
      };

      await this.props.addRecipe(recipe);

      this.props.history.push('/view_recipes');
    } catch (err) {
      if (err.response.data.error.message) {
        this.setState({
          loaded: true,
          uploadError: 'image is required',
          addingRecipe: false,
          ingredientError: '',
          titleError: '',
          instructionError: ''
        });
      }
    }
  }

  /**
   * @returns {obj} render
   */
  render() {
    let imageSource = null;
    if (this.state.image) {
      imageSource = this.state.image.preview;
    }
    const { uploadError } = this.state;
    return (
      <div className="main-userboard-body add-recipe-body">
        <div className="container">
          <div className="row">
            <div className="mx-auto col-sm-12 col-md-8 col-lg-8 col-xs mt-5">
              <h2 className="text-center"> Add Recipe</h2>
              <form onSubmit={this.addRecipe}>
                <Dropzone
                  onDrop={this.handleDrop}
                  multiple
                  accept="image/*"
                  style={dropZoneStyles}
                >

                  {this.state.image ?
                    <img className="card-img-top" style={{ height: '450px' }} alt="" src={imageSource} /> :
                    <div className="upload-recipe-img">
                      <div className="row justify-content-center">
                        <div className="col-12">
                          <p className="text-center" id="addImage">
                            <span className="h2"><i className="ion ion-camera" /></span>
                            <br />
                         Click to upload image
                          </p>
                        </div>
                      </div>
                    </div>
                 }
                </Dropzone>
                <div>
                  { uploadError &&
                  <span className="help-block error text-danger">
                    {uploadError}
                  </span>
                    }
                </div>

                <img src={`${this.state.imageURI}`} id="imagePreview" alt="" />
                <div className="form-group">
                  <label htmlFor="food">Recipe Name</label>
                  <input
                    onChange={this.onChange}
                    type="text"
                    name="title"
                    className="form-control form-control-lg"
                    placeholder="Enter Recipe name"
                  />
                  { this.state.titleError &&
                  <span className="help-block error text-danger">
                    {this.state.titleError}
                  </span>
                    }
                </div>

                <div className="form-group">
                  <label htmlFor="ingredient">Ingredients</label>
                  <input
                    onChange={this.onChange}
                    type="text"
                    name="ingredients"
                    className="form-control form-control-lg"
                    placeholder="Enter Ingredients"
                    id="ingredients"
                  />
                  { this.state.ingredientError &&
                    <span className="help-block error text-danger">
                      { this.state.ingredientError}
                    </span>
                    }
                </div>

                <div className="form-group">
                  <label htmlFor="instruction">Instructions</label>
                  <textarea
                    onChange={this.onChange}
                    className="form-control form-control-lg"
                    placeholder="Enter Instructions"
                    name="instructions"
                    id="instructions"
                  />
                  { this.state.instructionError &&
                  <span className="help-block error text-danger">
                    { this.state.instructionError}
                  </span>
                    }
                </div>
                <Loader
                  loaded={this.state.loaded}
                  lines={13}
                  length={30}
                  width={10}
                  radius={30}
                  corners={1}
                  rotate={0}
                  direction={1}
                  color="#000"
                  speed={2}
                  trail={60}
                  shadow={false}
                  hwaccel={false}
                  className="spinner"
                  zIndex={2e9}
                  top="70%"
                  left="50%"
                  scale={1.00}
                  loadedClassName="loadedContent"
                />

                <div className="form-group">
                  <input
                    type="submit"
                    className="btn btn-success"
                    disabled={this.state.addingRecipe}
                    value={this.state.addingRecipe ? 'Adding Recipe...' :
                     'Add Recipe'}
                  />
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}


AddRecipeForm.defaultProps = {
  addRecipe: {}
};

AddRecipeForm.propTypes = {
  addRecipe: PropTypes.func,
  history: PropTypes.shape({
    push: PropTypes.func
  }).isRequired
};


const mapStateToProps = state => ({
  recipes: state.recipes,
  loaded: state.loaded
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ addRecipe }, dispatch);

const ConnectedAddRecipeForm =
  connect(mapStateToProps, mapDispatchToProps)(AddRecipeForm);

export default ConnectedAddRecipeForm;
