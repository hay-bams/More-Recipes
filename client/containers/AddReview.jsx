import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Authenticate from '../auth/auth';
import { addReview } from '../actions/review';

/**
 * @class AddReview
 */
export class AddReview extends React.Component {
  /**
   * @returns {void} constructor
   */
  constructor() {
    super();
    this.addReview = this.addReview.bind(this);
    this.onChange = this.onChange.bind(this);
    this.clearForm = this.clearForm.bind(this);
    this.state = {
      review: '',
      reviewError: ''
    };
  }


  /**
  *
  * @param {obj} event
  * @returns {void} onChange
  */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   * @param {obj} event
   * @returns {void} clearForm
   */
  clearForm() {
    this.setState({ review: '' });
  }


  /**
   * @param {obj} event
   * @returns {void} addReview
   */
  addReview(event) {
    event.preventDefault();
    const userReview = {
      review: this.state.review
    };

    const errors = Authenticate.validateReview(userReview);

    if (errors.review !== '') {
      return this.setState({ reviewError: errors.review });
    }

    const userData = localStorage.getItem('userData');

    if (userData !== null) {
      const recipeId = parseInt(this.props.match.params.id, 10);
      this.props.addReview(userReview, recipeId);
      this.clearForm();
    } else {
      toastr.warning('you must be signed in');
      this.props.history.push('/signin');
    }
  }

  /**
   * @returns {void} render
   */
  render() {
    return (
      <div className="container">
        <div className="row review-form">
          <div className="col-lg-6">
            <h4>Add Review</h4>
            <form onSubmit={this.addReview}>
              <div className="form-group">
                <textarea
                  onChange={this.onChange}
                  className="form-control"
                  name="review"
                  value={this.state.review}
                />
                { this.state.reviewError &&
                  <span className="help-block error text-danger">
                    {this.state.reviewError}
                  </span>
                    }
              </div>

              <div className="form-group">
                <input type="submit" className="btn btn-success" />
              </div>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

AddReview.defaultProps = {
  addReview: () => {},
};

AddReview.propTypes = {
  addReview: PropTypes.func,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string
    })
  }).isRequired,
  history: PropTypes.shape({
    push: PropTypes.funct
  }).isRequired
};

const mapStateToProps = state => ({
  UserReviews: state.reviews
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ addReview }, dispatch);

const ConnectedAddReview =
  connect(mapStateToProps, mapDispatchToProps)(AddReview);

export default ConnectedAddReview;

