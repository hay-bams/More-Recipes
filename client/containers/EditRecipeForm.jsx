import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { editRecipe } from '../actions/actions';
import Authenticate from '../auth/auth';

/**
 * @class EditRecipeForm
 */
class EditRecipeForm extends React.Component {
  /**
   * @param {obj} props
   * @returns {void} constructor
   */
  constructor(props) {
    super(props);
    this.editRecipe = this.editRecipe.bind(this);
    this.onChange = this.onChange.bind(this);
    this.state = {
      recipe: {},
      errors: {}
    };
  }

  /**
   * @returns {void} componentWillMount
   */
  componentWillMount() {
    this.props.recipes.filter((recipe) => {
      if (recipe.id === parseInt(this.props.match.params.id, 10)) {
        this.setState({ recipe });
      }
    });
  }

  /**
   * @param {obj} event
   * @returns {void} onChange
   */
  onChange(event) {
    const recipe = {
      [event.target.name]: event.target.value
    };
    this.setState({ recipe });
  }

  /**
   * @param {obj} event
   * @returns {void} addRecipe
   */
  editRecipe(event) {
    event.preventDefault();
    const recipe = {
      title: event.target.title.value,
      ingredients: event.target.ingredients.value,
      instructions: event.target.instructions.value,
      image: 'imageURl'
    };

    let errors = Authenticate.validateAddRecipe(recipe);

    if (errors.title !== '' || errors.ingredients !== '' ||
     errors.instructions) {
      return this.setState({ errors });
    }

    this.props.editRecipe(recipe, this.props.match.params.id);
    // AddRecipeForm.clearForm(event);

    errors = {};
    this.setState({ errors });
  }

  /**
   * @returns {obj} render
   */
  render() {
    const { errors } = this.state;
    const { recipe } = this.state;
    return (
      <div className="main-userboard-body add-recipe-body">
        <div className="container">
          <div className="row">
            <div className="mx-auto col-sm-12 col-md-8 col-lg-6 col-xs mt-5">
              <h2 className="text-center"> Edit Recipe</h2>
              <form onSubmit={this.editRecipe}>
                <div className="form-group">
                  <label htmlFor="food">Recipe Name</label>
                  <input
                    type="text"
                    name="title"
                    className="form-control"
                    placeholder="Enter Recipe name"
                    value={recipe.title}
                    onChange={this.onChange}
                  />
                  { errors.title &&
                  <span className="help-block text-danger error">
                    {errors.title}
                  </span>
                    }
                </div>

                <div className="form-group">
                  <input type="file" />
                </div>

                <div className="form-group">
                  <label htmlFor="ingredient">Ingredients</label>
                  <input
                    type="text"
                    name="ingredients"
                    className="form-control"
                    placeholder="Enter Ingredients"
                    value={recipe.ingredients}
                    onChange={this.onChange}
                  />
                  { errors.ingredients &&
                    <span className="help-block text-danger error">
                      {errors.ingredients}
                    </span>
                    }
                </div>

                <div className="form-group">
                  <label htmlFor="instruction">Instructions</label>
                  <textarea
                    className="form-control"
                    placeholder="Enter Instructions"
                    name="instructions"
                    id="instructions"
                    value={recipe.instructions}
                    onChange={this.onChange}
                  />
                  { errors.instructions &&
                  <span className="help-block text-danger error">
                    {errors.instructions}
                  </span>
                    }
                </div>

                <div className="form-group">
                  <input
                    type="submit"
                    className="btn btn-success"
                    value="Update Recipe"
                  />
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>

    );
  }
}


EditRecipeForm.defaultProps = {
  editRecipe: {}
};

EditRecipeForm.propTypes = {
  editRecipe: PropTypes.func,
  recipes: PropTypes.arrayOf(PropTypes.shape({
    upvotes: PropTypes.number,
    downvotes: PropTypes.number,
    id: PropTypes.number,
    title: PropTypes.string,
    image: PropTypes.string,
    instructions: PropTypes.string,
    ingredients: PropTypes.string,
    userId: PropTypes.number,
    createdAt: PropTypes.string,
    updatedAt: PropTypes.string
  })).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string
    })
  }).isRequired
};


const mapStateToProps = state => ({
  recipes: state.userRecipes
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ editRecipe }, dispatch);


export default connect(mapStateToProps, mapDispatchToProps)(EditRecipeForm);
