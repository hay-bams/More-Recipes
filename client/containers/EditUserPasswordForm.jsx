import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { editUserPassword } from '../actions/user';
import Authenticate from '../auth/auth';

/**
 * @class EditRecipeForm
 */
export class EditUserPasswordForm extends React.Component {
  /**
   * @param {obj} props
   * @returns {void} constructor
   */
  constructor(props) {
    super(props);
    this.editPassword = this.editPassword.bind(this);
    this.onChange = this.onChange.bind(this);
    this.state = {
      password: '',
      confirmPassword: '',
      passwordError: '',
      passwordMatchError: ''
    };
  }

  /**
   * @param {obj} event
   * @returns {void} onChange
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   * @param {obj} event
   * @returns {void} addRecipe
   */
  editPassword(event) {
    event.preventDefault();
    const user = {
      password: this.state.password
    };

    const errors = Authenticate.validatePassword(user, this.state.confirmPassword);

    if (errors.password !== '' || errors.confirmPassword !== '') {
      return this.setState({
        passwordError: errors.password,
        passwordMatchError: errors.confirmPassword
      });
    }
    this.props.editUserPassword(user, this.props.match.params.id);
  }

  /**
   * @returns {obj} render
   */
  render() {
    return (
      <div className="container-fluid main-register-container">
        <div className="row overlay">
          <div className="col-sm-8 col-md-5 mx-auto">
            <div className="card  mt-5 card-form">
              <div className="card-header text-center">
                <h2 className="text-center"> Edit Password</h2>
              </div>
              <span>{this.props.errorMsg}</span>
              <div className="card-body">
                <form onSubmit={this.editPassword}>

                  <div className="form-group">
                    <input
                      type="password"
                      name="password"
                      className="form-control"
                      placeholder="Enter New Password"
                      value={this.state.password}
                      onChange={this.onChange}
                    />
                    { this.state.passwordError &&
                    <span className="help-block error text-danger">
                      {this.state.passwordError}
                    </span>
                      }
                  </div>

                  <div className="form-group">
                    <input
                      type="password"
                      name="confirmPassword"
                      className="form-control"
                      placeholder="Confirm Password"
                      value={this.state.confirmPassword}
                      onChange={this.onChange}
                    />
                    { this.state.passwordMatchError &&
                    <span className="help-block error text-danger">
                      {this.state.passwordMatchError}
                    </span>
                      }
                  </div>

                  <div className="form-group">
                    <input
                      type="submit"
                      value="Update Password"
                      className="btn btn-success btn-block"
                    />
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}


EditUserPasswordForm.defaultProps = {
  editUserPassword: {},
  errorMsg: ''
};

EditUserPasswordForm.propTypes = {
  editUserPassword: PropTypes.func,
  errorMsg: PropTypes.string,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string
    })
  }).isRequired
};


const mapDispatchToProps = dispatch =>
  bindActionCreators({ editUserPassword }, dispatch);

const ConnectedEditUserPasswordForm =
  connect(null, mapDispatchToProps)(EditUserPasswordForm);

export default ConnectedEditUserPasswordForm;

