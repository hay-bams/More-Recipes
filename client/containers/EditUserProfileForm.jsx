import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { editUserProfile } from '../actions/actions';
import Authenticate from '../auth/auth';

/**
 * @class EditRecipeForm
 */
class EditUserProfileForm extends React.Component {
  /**
   * @param {obj} props
   * @returns {void} constructor
   */
  constructor(props) {
    super(props);
    this.editProfile = this.editProfile.bind(this);
    this.onChange = this.onChange.bind(this);
    this.state = {
      errors: {},
      user: {}
    };
  }

  /**
   * @returns {void} componentWillMount
   */
  componentWillMount() {
    const { user } = this.props.userData;
    this.setState({ user });
  }

  /**
   * @param {obj} event
   * @returns {void} onChange
   */
  onChange(event) {
    const user = {
      [event.target.name]: event.target.value
    };
    this.setState({ user });
  }

  /**
   * @param {obj} event
   * @returns {void} addRecipe
   */
  editProfile(event) {
    event.preventDefault();
    const user = {
      firstName: event.target.firstName.value,
      lastName: event.target.lastName.value,
      email: event.target.email.value,
      password: event.target.password.value
    };

    let errors = Authenticate.validateUserSignup(user);

    if (errors.email !== '' || errors.password !== '' ||
       errors.firstName !== '' || errors.lastName !== '') {
      return this.setState({ errors });
    }
    this.props.editUserProfile(user, this.props.match.params.id);
    errors = {};
    this.setState({ errors });
  }

  /**
   * @returns {obj} render
   */
  render() {
    const { errors } = this.state;
    const { user } = this.state;
    return (
      <div className="container-fluid main-register-container">
        <div className="row overlay">
          <div className="col-sm-8 col-md-5 mx-auto">
            <div className="card  mt-5 card-form">
              <div className="card-header text-center">
                <h2 className="text-center"> Edit User Profile</h2>
              </div>
              <span>{this.props.errorMsg}</span>
              <div className="card-body">
                <form onSubmit={this.editProfile}>
                  <div className="form-group">
                    <input
                      type="text"
                      name="firstName"
                      className="form-control"
                      placeholder="Enter First name"
                      value={user.firstName}
                      onChange={this.onChange}
                    />
                    { errors.firstName &&
                    <span className="help-block error text-danger">
                      {errors.firstName}
                    </span>
                      }
                  </div>

                  <div className="form-group">
                    <input
                      type="text"
                      name="lastName"
                      className="form-control"
                      placeholder="Enter Last name"
                      value={user.lastName}
                      onChange={this.onChange}
                    />
                    { errors.lastName &&
                    <span className="help-block error text-danger">
                      {errors.lastName}
                    </span>
                      }
                  </div>

                  <div className="form-group">
                    <input
                      type="email"
                      name="email"
                      className="form-control"
                      placeholder="Enter Email"
                      value={user.email}
                      onChange={this.onChange}
                    />
                    { errors.email &&
                    <span className="help-block error text-danger">
                      {errors.email}
                    </span>
                      }
                  </div>

                  <div className="form-group">
                    <input
                      type="password"
                      name="password"
                      className="form-control"
                      placeholder="Enter Password"
                      value={user.password}
                      onChange={this.onChange}
                    />
                    { errors.password &&
                    <span className="help-block error text-danger">
                      {errors.password}
                    </span>
                      }
                  </div>

                  <div className="form-group">
                    <input
                      type="submit"
                      value="Update Profile"
                      className="btn btn-success btn-block"
                    />
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}


EditUserProfileForm.defaultProps = {
  editUserProfile: {},
  userData: {},
  errorMsg: ''
};

EditUserProfileForm.propTypes = {
  editUserProfile: PropTypes.func,
  userData: PropTypes.shape({
    user: PropTypes.shape({
      email: PropTypes.string,
      firstName: PropTypes.string,
      id: PropTypes.number,
      lastName: PropTypes.string
    })
  }),
  errorMsg: PropTypes.string,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string
    })
  }).isRequired
};


const mapStateToProps = state => ({
  userData: state.userData,
  errorMsg: state.errors.signUpError
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ editUserProfile }, dispatch);


export default connect(mapStateToProps, mapDispatchToProps)(EditUserProfileForm);
