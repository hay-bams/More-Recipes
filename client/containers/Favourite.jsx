import React from 'react';
import { Link } from 'react-router-dom';
import _ from 'lodash';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { confirmAlert } from 'react-confirm-alert';
// import 'react-confirm-alert/src/react-confirm-alert.css';
import ReactPaginate from 'react-paginate';
import { bindActionCreators } from 'redux';
import { getFavouriteRecipes, deleteFavoriteRecipe } from '../actions/recipe';

/**
 * @class FavouriteRecipes
 */
export class FavouriteRecipes extends React.Component {
  /**
   * @returns {void} constructor
   */
  constructor() {
    super();
    this.handlePageClick = this.handlePageClick.bind(this);
    this.renderFavouriteRecipes = this.renderFavouriteRecipes.bind(this);
    this.onDelete = this.onDelete.bind(this);
    this.showModal = this.showModal.bind(this);
  }
  /**
   * @return {void} componentWillMount
   */
  componentWillMount() {
    this.props.getFavouriteRecipes(1);
  }

  /**
   * @param {number} recipeId
   * @return {void} onDelete
   */
  onDelete(recipeId) {
    this.props.deleteFavoriteRecipe(recipeId);
  }

  /**
   * @param {obj} page
   * @returns {void} handlePageClick
   */
  handlePageClick(page) {
    const selected = page.selected + 1;
    this.props.getFavouriteRecipes(selected);
  }

  /**
   *
   * @param {obj} event
   * @returns {obj} showModal
   */
  showModal(event) {
    const recipeId = parseInt(event.target.id, 10);
    return (confirmAlert({
      title: 'Confirm Delete',
      message: 'Are you sure to delete this recipe ?',
      confirmLabel: 'Yes',
      cancelLabel: 'No',
      onConfirm: () => this.onDelete(recipeId),
      onCancel: () => '',
    })
    );
  }

  /**
   * @returns {obj} renderRecipe
   */
  renderFavouriteRecipes() {
    const { favouriteRecipes } = this.props;
    return favouriteRecipes.length === 0 ?
      <div className="mx-auto">
        <p className="text-center">
        You do not have any favorite recipe at the moment.
        </p>
      </div> :
      favouriteRecipes.map(favouriteRecipe => (
        <div
          className="col-12 col-sm-6 col-md-6 col-lg-4 recipes"
          key={favouriteRecipe.id}
        >
          <div className="card recipe-card" style={{ border: 'none' }}>
            <img
              className="card-img-top img-fluid img-recipe"
              src={favouriteRecipe.image}
              alt="Card  cap"
              style={{ height: `${200}px` }}
            />
            <div className="card-body mx-auto">
              <h4 className="card-title text-center">{favouriteRecipe.title}</h4>
              <p className="card-text">
                <span className="row">
                  <span href="#" className="text-success">
                    <i
                      className="fa fa-thumbs-up col-4"
                      aria-hidden="true"
                    />{favouriteRecipe.upvotes}
                  </span>
                  <span href="#" className="text-info">
                    <i className="fa fa-eye text-info col-4" aria-hidden="true" /> {favouriteRecipe.views}
                  </span>
                  <span href="#" className="text-danger">
                    <i
                      className="fa fa-thumbs-down col-4"
                      aria-hidden="true"
                    />{favouriteRecipe.downvotes}
                  </span>
                </span>
              </p>
              <Link to={`/details/${favouriteRecipe.id}`} className="btn btn-outline-info" style={{ marginLeft: `${5}px` }}>
               View
              </Link>

              <button className="btn btn-outline-info ml-2" id={favouriteRecipe.id} onClick={this.showModal}>
                Remove
              </button>
            </div>
          </div>
        </div>


      ));
  }

  /**
   * @returns {void} render
   */
  render() {
    return (
      <div className="main">
        <div className="container">
          <div className="row">
            <div className="col-sm">
              <h2 className="text-center catalogue">My Favorite Recipes</h2>
              <h3 className="text-center">{this.message}</h3>
            </div>
          </div>
          <div className="row">
            {this.renderFavouriteRecipes()}
          </div>

          <div className="row">
            <div className="col-md-8 ml-auto">
              { this.props.favouriteRecipes.length > 0 ?
                <ReactPaginate
                  previousLabel="Previous"
                  nextLabel="Next"
                  breakLabel={<a href="">...</a>}
                  breakClassName="page-link"
                  pageCount={this.props.pages}
                  onPageChange={this.handlePageClick}
                  containerClassName="pagination pagination-md"
                  pageLinkClassName="page-link"
                  nextLinkClassName="page-link"
                  previousLinkClassName="page-link"
                  disabledClassName="disabled"
                  pageClassName="page-item"
                  previousClassName="page-item"
                  nextClassName="page-item"
                  activeClassName="active"
                  subContainerClassName="pages pagination"
                /> : ''
          }
            </div>
          </div>

        </div>
      </div>
    );
  }
}

FavouriteRecipes.defaultProps = {
  pages: 1
};

FavouriteRecipes.propTypes = {
  getFavouriteRecipes: PropTypes.func.isRequired,
  deleteFavoriteRecipe: PropTypes.func.isRequired,
  favouriteRecipes: PropTypes.arrayOf(PropTypes.shape({
    upvotes: PropTypes.number,
    downvotes: PropTypes.number,
    id: PropTypes.number,
    title: PropTypes.string,
    image: PropTypes.string,
    instructions: PropTypes.string,
    ingredients: PropTypes.string,
    userId: PropTypes.number,
    createdAt: PropTypes.string,
    updatedAt: PropTypes.string
  })).isRequired,
  pages: PropTypes.number
};

const mapStateToProps = state => ({
  favouriteRecipes: state.favouriteRecipes.rows,
  pages: state.favouriteRecipes.pages,
  errorMsg: state.errors.addFavoriteError
});

const mapDispatchToProps = dispatch => bindActionCreators(
  { getFavouriteRecipes, deleteFavoriteRecipe },
  dispatch
);

const ConnectedFavouriteRecipe =
  connect(mapStateToProps, mapDispatchToProps)(FavouriteRecipes);


export default ConnectedFavouriteRecipe;

