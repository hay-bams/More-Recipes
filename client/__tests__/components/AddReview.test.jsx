import React from 'react';
import { shallow } from 'enzyme';
import { AddReview } from '../../containers/AddReview';

describe('', () => {
  const props = {
    addReview: jest.fn(),
    history: {
      push: jest.fn()
    },
    match: {
      params: {
        id: '1'
      }
    }
  };

  test('it should render AddReviewForm correctly', () => {
    const wrapper = shallow(<AddReview {...props} />);
    expect(wrapper.find('form').exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  test('it should render error for empty review in form submission', () => {
    const wrapper = shallow(<AddReview {...props} />);
    wrapper.instance().setState({
      review: ''
    });
    wrapper.update();
    wrapper.find('form').simulate('submit', {
      preventDefault: () => {}
    });
    expect(wrapper.state('reviewError').length).toBeGreaterThan(0);
  });


  test('should set review state on input change', () => {
    const value = 'some review';
    const name = 'review';
    const wrapper = shallow(<AddReview {...props} />);
    wrapper.find('textarea').simulate('change', {
      target: { name, value },
      preventDefault: () => {}
    });
    expect(wrapper.state('review')).toBe(value);
  });

  test('should call addReview prop for valid form submission', () => {
    const review = {
      review: 'some reviews'
    };
    const wrapper = shallow(<AddReview {...props} />);
    wrapper.instance().setState({
      review: 'some reviews'
    });
    wrapper.update();
    wrapper.find('form').simulate('submit', {
      preventDefault: () => {}
    });

    expect(props.addReview).toHaveBeenLastCalledWith(review, 1);
  });
});

