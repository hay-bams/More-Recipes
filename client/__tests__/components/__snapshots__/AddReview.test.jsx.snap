// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render AddReview form 1`] = `
<div
  className="container"
>
  <div
    className="row review-form"
  >
    <div
      className="col-lg-6"
    >
      <h4>
        Add Review
      </h4>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <textarea
            className="form-control"
            name="review"
            onChange={[Function]}
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="btn btn-success"
            type="submit"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
